### PR TITLE
Add summarisation part of the NorGLM NO-Multi-QA-Sum dataset

### PR DESCRIPTION
**Pull request by viggo-gascou**
Added the [NorGLM NO-Multi-QA-Sum dataset](https://huggingface.co/datasets/NorGLM/NO-Multi-QA-Sum)

I have tried to use the dataset by running
scandeval --model mhenrichsen/danskgpt-tiny --dataset norglm-multi-sum. I do not seem to get any errors, although it is running rather slow (I think this is a standing issue with my machine - have not tried to debug further yet).

Fixes part of #536

The dataset is somewhat small and consists of 467 samples with the following splits:

- 147 training samples
- 64 validation samples
- 256 test samples

Working on adding the question and answer part of the same dataset as well.

### CODE REVIEW

**Comment by viggo-gascou** [comment]
From inspection of the question and answer part of the dataset it seems that not many of the answers can directly be found in the article or the summary (only 15). So it probably does not make sense to include the question and answer part of the dataset unless we do some reformulation of all the answers to ensure the answer can be found in the context. How do you think I should proceed @saattrupdan?

**Comment by viggo-gascou** [comment]
Hmm it looks like there are a few duplicate articles as well but with different summaries and questions see them [here](https://huggingface.co/datasets/NorGLM/NO-Multi-QA-Sum/embed/sql-console/default/train?sql_console=true&views%5B%5D=train&sql=--+The+SQL+console+is+powered+by+DuckDB+WASM+and+runs+entirely+in+the+browser.%0A--+Get+started+by+typing+a+query+or+selecting+a+view+from+the+options+below.%0ASELECT+*+FROM+train+WHERE+article+IN+%28+SELECT+article+FROM+train+GROUP+BY+article+HAVING+COUNT%28*%29+%3E+1+%29+ORDER+BY+article+). Additionally, I found an "article" which was just a bunch of questions and answers from a quiz of some sort see [here](https://huggingface.co/datasets/NorGLM/NO-Multi-QA-Sum/embed/sql-console/default/train?sql_console=true&views%5B%5D=train&sql=--+The+SQL+console+is+powered+by+DuckDB+WASM+and+runs+entirely+in+the+browser.%0A--+Get+started+by+typing+a+query+or+selecting+a+view+from+the+options+below.%0ASELECT+*+FROM+train+WHERE+%22Unnamed%3A+0%22+%3D+%27359%27&sql_row=0). If I remove the duplicate articles and the non-article entry the total number of samples drops to 451 (that is keeping one of the duplicate articles). We could remove all duplicates or just choose a random one, not sure which is better?

**Comment by saattrupdan** [comment]
> From inspection of the question and answer part of the dataset it seems that not many of the answers can directly be found in the article or the summary (only 15). So it probably does not make sense to include the question and answer part of the dataset unless we do some reformulation of all the answers to ensure the answer can be found in the context. How do you think I should proceed @saattrupdan?

This is similar to the Icelandic QA dataset that @oliverkinch looked at ([link](https://github.com/ScandEval/ScandEval/pull/564)) - you can try to mimic his approach with reformulating the answers. But it should probably be in a separate PR, as this one focuses on the summarisation part.

**Comment by saattrupdan** [comment]
> Hmm it looks like there are a few duplicate articles as well but with different summaries and questions see them [here](https://huggingface.co/datasets/NorGLM/NO-Multi-QA-Sum/embed/sql-console/default/train?sql_console=true&views%5B%5D=train&sql=--+The+SQL+console+is+powered+by+DuckDB+WASM+and+runs+entirely+in+the+browser.%0A--+Get+started+by+typing+a+query+or+selecting+a+view+from+the+options+below.%0ASELECT+*+FROM+train+WHERE+article+IN+%28+SELECT+article+FROM+train+GROUP+BY+article+HAVING+COUNT%28*%29+%3E+1+%29+ORDER+BY+article+). Additionally, I found an "article" which was just a bunch of questions and answers from a quiz of some sort see [here](https://huggingface.co/datasets/NorGLM/NO-Multi-QA-Sum/embed/sql-console/default/train?sql_console=true&views%5B%5D=train&sql=--+The+SQL+console+is+powered+by+DuckDB+WASM+and+runs+entirely+in+the+browser.%0A--+Get+started+by+typing+a+query+or+selecting+a+view+from+the+options+below.%0ASELECT+*+FROM+train+WHERE+%22Unnamed%3A+0%22+%3D+%27359%27&sql_row=0). If I remove the duplicate articles and the non-article entry the total number of samples drops to 451 (that is keeping one of the duplicate articles). We could remove all duplicates or just choose a random one, not sure which is better?

I think we just choose a random one of them (with a random seed, of course). With 451 samples we could have a test set of 256, use 64 for validation and the remaining 131 for training, say.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Added
+- Added the summarisation part of the Norwegian NorGLM multi-task human annotated
+  dataset NO-Multi-QA-Sum. This dataset is part of the It is a part of NLEBench
+  Norwegian benchmarks. It has been marked as `unofficial` for now.
```
Comment (saattrupdan) [inline CHANGELOG.md]: Typo:

```suggestion
  dataset NO-Multi-QA-Sum (`norglm-multi-sum`). This dataset is part of the NLEBench
  Norwegian benchmarks. It has been marked as `unofficial` for now.
```

File: src/scripts/create_norglm_multisum.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    dataset = dataset.rename_columns(
+        column_mapping=dict(article="text", summary="target_text")
+    )
+
+    df = dataset.to_pandas()
+    assert isinstance(df, pd.DataFrame)
+
+    # Only work with samples where the text is not very large or small
+
```
Comment (saattrupdan) [inline src/scripts/create_norglm_multisum.py:32]:
```suggestion
```

File: src/scripts/create_norglm_multisum.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    # Create test split
+    test_size = 256
+    filtered_df = df[~df.index.isin(val_df.index)]
+    test_df = filtered_df.sample(n=test_size, random_state=4242)
+
+    # Create train split
+    train_size = 147
+    filtered_df = filtered_df[~filtered_df.index.isin(test_df.index)]
+    train_df = filtered_df.sample(n=train_size, random_state=4242)
```
Comment (saattrupdan) [inline src/scripts/create_norglm_multisum.py]: No need to specify the train size, as it is simply the rest:

```suggestion
    train_df = filtered_df[~filtered_df.index.isin(test_df.index)]
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]
Looks good! Just a couple of smaller things.

**Comment by viggo-gascou** [comment]
fixed now, commit didn't update for some reason 😅

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Pull request merged**